### DISCUSSION TITLE
`base16` theming using `tinty`

### DISCUSSION THREAD

**Discussion by salahoued** [labwc/labwc#2454] [Show and tell]
Configuration for [`tiny`](https://github.com/tinted-theming/tinty) a `Rust`-based, `pywal` like tool to generate color schemes from the wallpaper (image).

The theme is based on a `breeze-dark` theme for `openbox`
### Configuration snippet to put in `~/.config/tinted-theming/tinty/config.toml`
```
[[items]]
name = "base16-labwc"
path = "~/.local/share/tinted-theming/tinty/repos/base16-labwc"
themes-dir = "colors"
hook = "cp -r %f ~/.local/share/themes/breeze-dark-ob-tinted/openbox-3/themerc && labwc -r"
supported-systems = ["base16"]
```
### The `Breeze-dark` based `openbox` theme
[breeze-dark-ob-tinted.zip](https://github.com/user-attachments/files/18256293/breeze-dark-ob-tinted.zip)

### The `labwc` configuration and template for `tinty`
Extract/copy to `~/.local/share/tinted-theming/tinty/repos`
[base16-labwc.zip](https://github.com/user-attachments/files/18256295/base16-labwc.zip)

### Screenshots
![wayshot_2024-12-27-01:12:06](https://github.com/user-attachments/assets/2478256c-d320-4a2f-b5df-aebe641e219b)

![wayshot_2024-12-27-01:15:22](https://github.com/user-attachments/assets/38790e97-3d5d-4486-920d-af75866cdcc2)

#### The script used to select an image (depends on `fzf`, `fd`, `sd` and `imagemagick`)
[tinted-wall.sh.txt](https://github.com/user-attachments/files/18256346/tinted-wall.sh.txt)

If you're using `waypaper`, you can run this script: [tinted-waypaper.sh.txt](https://github.com/user-attachments/files/18256414/tinted-waypaper.sh.txt) as a `post_command` in `waypaper`'s configuration file like so:
```
[Settings]
language = en
...
backend = swaybg
fill = fill
sort = name
color = #ffffff
subfolders = True
show_hidden = False
show_gifs_only = False
post_command = tinted-waypaper.sh $wallpaper
number_of_columns = 3
...
```

The script used to set the wallpaper for the screenlocker  used in the previous script `tinted-wall.sh`
[so-bg-locker.sh.txt](https://github.com/user-attachments/files/18256349/so-bg-locker.sh.txt)

To keep the chosen wallpaper after a logout/restart, consider replacing your wallpaper in `~/.config/labwc/autostart` or any where you configured you wallpaper with something like this:
For `swaybg`
`swaybg -o "*" -m fill -i "$(xdg-user-dir PICTURES)/current-wallpaper"  >/dev/null 2>&1 &`

Hope you like it

**Comment by johanmalm** [comment]
Hey. Very nice!
